modules/ItemLevel.lua

```diff
@@ -112,68 +112,64 @@ end
 
 function mod:UpdateButton(event, button)
 	local settings = self.db.profile
-	local link = button:GetItemLink()
 	local text = texts[button]
-
-	if link then
-		local objectTable = { strsplit(":", string.match(link, "|H(.-)|h")) }
-		-- example objectTable for a battle pet:
-		-- [1]="battlepet", -- item type
-		-- [2]="242",
-		-- [3]="25",        -- pet level
-		-- [4]="3",
-		-- [5]="1319",
-		-- [6]="305",
-		-- [7]="289",
-		-- [8]="0000000000000000",
-		-- [9]="30409"
-		if objectTable[1] == "battlepet" then
-			if settings.showBattlePetLevels then
-				local battlepetlevel = objectTable[3]
-				if not text then
-					text = CreateText(button)
-				end
-				text:SetText(battlepetlevel)
-				return text:Show()
-			elseif text then
-				return text:Hide()
-			else
-				return
+	--Integration with SyLevel. If useSyLevel then let SyLevel handle item level logic.
+	if SyLevel then	
+		if settings.useSyLevel then
+			if text then
+				text:Hide()
 			end
+			SyLevel:CallFilters('Adibags', button, link)
```

Comment:
link isn't defined yet

Reply:
Good catch.  I installed SyLevel for testing and found that it has several errors. It hasn't been updated since 2018 and the api changes for Shadowlands cause several errors.  Support for SyLevel should probably be removed altogether.